pkg/handlers/generic/lifecycle/cni/cilium/handler.go

```diff
@@ -175,9 +177,16 @@ func (c *CiliumCNI) apply(
 			)
 			return
 		}
+
+		helmValuesConfigmapRef := c.config.helmAddonConfig.defaultValuesTemplateConfigMapName
+		if cniVar.Values != nil && cniVar.Values.SourceRef != nil {
+			helmValuesConfigmapRef = cniVar.Values.SourceRef.Name
+			defaultNamespace = cluster.Namespace
```

Comment:
```suggestion
			// use the cluster's namespace since Values.SourceRef is always a LocalObjectReference
			defaultNamespace = cluster.Namespace
```